### PR TITLE
Enchancing output

### PR DESCRIPTION
**Pull request by jinder1s**
- Added github_url for each toggle annotation
- Changed how name is output
- removed html renderer
- rename idas

To make PR easier, I plan on blacking everything after the PR has been mostly approved. @ reviewer, let me know if you think this is a bad idea.

Jenkin job PR that uses this: https://github.com/edx/jenkins-job-dsl-internal/pull/396/files
The resulting output can be found in artifacts here: https://tools-edx-jenkins.edx.org/job/Feature-Toggle-Report-Generator/job/publish-feature-toggle-report/

### CODE REVIEW

**Comment by robrap** [comment]
@jinder1s: Regarding CSVs, I have the following general comments.  Let's see what we should add additional subtasks for.  Thanks.
1. `state_languages` for WaffleFlags seems pretty useless, but we can always iterate on that.
2. `full_report.csv` could be renamed to `all_toggles_report.csv`
3. Where is the CourseWaffleFlag and info?  This is complicated, but it was done. 
4. You may want a settings_report.csv for consistency.
5. I'm not sure that alphabetical order is the best option.  It might be ok once we add back the computed column, especially once CourseWaffleFlag is back.

File: scripts/README.rst [hunk 1]
```diff
@@ -116,7 +123,23 @@ IMPORTANT: Example of data structure:
...
+
+The script can also take a yaml file as configuration, though flags will overwrite things in configuration. 
+
+Valid keys in configuration file:
+    - env: list the envs you want included in report
+    - toggle_type:Ld list the toggle types you want in report
+    - show_state: set to true if you want to output toggles states in report
+    - ida: list configurations settings for each ida
+
+valid keys in ida settings:
```
[the remark this answers is not in the document]
Comment (jinder1s) [reply scripts/README.rst:140]: I meant sub keys under ida key in yaml file. I will change this to look better. So in dict form
 {ida:{rename:name, github_url:url}}

File: scripts/feature_toggle_report_generator.py [hunk 2]
```diff
@@ -22,54 +28,88 @@
...
-        if os.path.isdir(os.path.join(toggle_data_dir, path)) and env_name:
-            envs_data_paths.append((os.path.join(toggle_data_dir, path), env_name))
+        env_name_search = env_name_pattern.search(path)
+        if os.path.isdir(os.path.join(toggle_data_dir, path)) and env_name_search:
+            env_data_paths.append((os.path.join(toggle_data_dir, path), env_name_search.group('env')))
 
     total_info = {}
     for env_data_path, env_name in env_data_paths:
-        if env is not None and env_name != env:
+        if requested_envs and env_name not in requested_envs:
```
[the remark this answers is not in the document]
Comment (jinder1s) [reply scripts/feature_toggle_report_generator.py:101]: The check for requested_envs makes sure there is something in requested_envs. If there is nothing in requested_envs, its assumed every env is valid for output

[the remark this answers is not in the document]
Comment (jinder1s) [reply scripts/feature_toggle_report_generator.py:101]: I plan on adding these comments: `# if an env is specified in requested_envs, filter out everyother env
        # if no env is specified, assume all envs are valid`

File: scripts/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (robrap) [reply scripts/README.rst:140]: Probably fine as-is if you just provide an example config file.

File: scripts/feature_toggle_report_generator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (robrap) [reply scripts/feature_toggle_report_generator.py:101]: I don't think you need a comment.  My brain just wasn't working.  You can just leave this.  Thanks.

**Comment by jinder1s** [comment]
What I changed based on your suggestion: 
- full_report.csv is will now be named all_toggles_report.csv
- Adding CourseWaffleFlag to flag_report.csv
- adding settings report

We should talk about rest

**Comment by robrap** [comment]
@jinder1s: You should first update the release version and CHANGELOG entry for 0.3.0.  See https://github.com/edx/edx-toggles/pull/41/files, where Régis started fixing this.  I'll have Régis update his to 0.4.0.

**Pull request merged**